### PR TITLE
Reader: hide discover pick byline

### PR DESCRIPTION
**Pull request by jhnstn**
This hides the byline for discover posts. 

The change request came out of a discussion around content attribution for these posts.  The current design leads too many to attribute the editor as the original author of the content.  The decision for now is to remove the editor byline from discovery posts.

### Screens

**Before**
screen shot 2016-08-01 at 10 51 26 am

**After**
screen shot 2016-08-01 at 10 52 56 am

### Testing

``` npm run test-client client/reader/post-byline ```

Test live: https://calypso.live/?branch=update/reader/hide-discover-byline

### CODE REVIEW

File: client/reader/post-byline/test/index.jsx [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+			} );
+
+			it( 'does not render if the post author does not have a name', () => {
+				const author = omit( post.author, 'name' );
+				const nameLessByline = shallow( <PostByline post={ Object.assign( {}, post, { author } ) } /> );
+				assert.lengthOf( nameLessByline.find( authorSelector ), 0 );
+			} );
+
+			it( 'does not render if the post is a discover pick', () => {
+				const disoverPick = shallow( <PostByline post={ post } isDiscoverPost={ true } /> );
```
Comment (bluefuton) [inline client/reader/post-byline/test/index.jsx]: Tiny typo here (disoverPick -> discoverPick)

**Comment by folletto** [comment]
Can you add before/after screenshots please? :)

**Comment by jhnstn** [comment]
@folletto screenshots added to the description

**Comment by folletto** [comment]
Given the discussion you had, feels a good choice. 👍

**Comment by jancavan** [comment]
This actually feels like it lessens the confusion. LGTM :shipit:

**Comment by fraying** [comment]
I love this.

**Comment by jhnstn** [comment]
After looking at the screens I had this idea:

**Before**
screen shot 2016-08-01 at 11 04 10 am

**After** 
discover-byline

This gist is moving the source attribution up to where the editor byline was.

cc @folletto @jancavan @fraying

**Comment by jancavan** [comment]
@jhnstn I think that might be confusing because the excerpt was not written by Megan. It's written by one of the Discover editors, which we are removing in this PR.

**Comment by jhnstn** [comment]
@jancavan good point. Leaving the changes as is

File: client/reader/post-byline/index.jsx [hunk 2]
```diff
@@ -1,40 +1,63 @@
...
+	recordTrackForPost,
+	recordPermalinkClick
+ } from 'reader/stats';
+
+class PostByline extends React.Component {
+
+	constructor( ) {
+		super( );
+		this.recordTagClick = this.recordTagClick.bind( this );
+		this.recordAuthorClick = this.recordAuthorClick.bind( this );
```
Comment (omarjackman) [inline client/reader/post-byline/index.jsx]: I'm curious about why your binding these two functions to `this` when they should are already be bound correctly. Maybe I'm missing something. If its to satisfy some condition where `this` is lost when passing the function as a callback somewhere else then just passing `() => this.recordTagClick()` should suffice.

Comment (jhnstn) [inline client/reader/post-byline/index.jsx]: React dropped autobinding as part of the move to ES6 classes.  The `onClick` event will find `this.recordTagClick` but within the function call `this.props` will be undefined. 

It's possible to do the binding inside the JSX but there is a [performance hit](https://github.com/yannickcr/eslint-plugin-react/issues/473)  doing that. 

It's possible to sweeten the syntax with:
```
constructor() {
  ...
  this._recordTagClick = () => this.recordTagClick;
  ...
}
```
But I prefer the explicit bind calls.

Looks like es7 has some object property scoping goodness but we're not using the es7 babel preset

**Comment by bluefuton** [comment]
Good to go I reckon 🚢

**Pull request merged**